Feature/VIMPPI controller

lvjonok opened the pull request:
* feat: new controllers for the benchmarks

* wip: going to gentoo

* feat: collect distribution data

* feat: more controllers

* fix: draw table

* feat: all controllers

* feat: remove unrelated stuff and update dependencies

* fix: update nvidia container

* fix: restore file

* fix: remove unrelated

* fix: restore formatting

---------

fwiebe commented:
Hi @lvjonok  , thank you for your submission. All necessary files are provided, the controller runs without errors, and the scores are as reported.

Could you please address the following remarks before the pull request can be merged to the main repository?

- The controller is name is vimppi (one m, double p), right? The files in the leaderboard folders are called con_vimmpi (double m, one p) at the moment.
- Please remove the leaderboard/pendubot/simulation_v3/.DS_Store file
- Could you add small readme files in the leaderboard/(acrobot/pendubot)/simulation_v3/readmes folders which briefly describe your controller? This will be linked on the leaderboards.
- Please decide for one controller for each system and remove the others from the leaderbaord folders. It is fine to try out and switch to another one in the hardware phase, but for the evaluations please submit only one.
- Would you place your plant file (src/python/double_pendulum/controller/vimppi/plant.py) and you integrator file (src/python/double_pendulum/controller/vimppi/integrator.py) in the model folder (https://github.com/dfki-ric-underactuated-lab/double_pendulum/tree/main/src/python/double_pendulum/model) and the simulation folder (https://github.com/dfki-ric-underactuated-lab/double_pendulum/tree/main/src/python/double_pendulum/simulation) respectively? Then they can easily used by others if desired.
- Executing the controller takes very long on my PC ( ~80 minutes for one leaderboard run). Run without GPU. Is this expected? Could it be sped up?

lvjonok commented:
Hi @fwiebe !

We have addressed the issues you have raised in the message, please review once more. We can do more iterations on your comments.

Regarding the speed of execution:
Yes, indeed, the CPU-only execution will take a tremendous amount of time. Our algorithm is expected to run on GPU. As a change right now, we have altered the default FROM command in the Dockerfile so it uses nvidia-containers under the hood.

Please see this for the setup: https://docs.nvidia.com/datacenter/cloud-native/container-toolkit/latest/install-guide.html

When docker is built, you can run the docker via:

`docker run -it --runtime=nvidia --gpus all ...`

Looking forward your further feedback!

fwiebe commented:
Thanks @lvjonok ,

Thanks for addressing the comments. With a GPU it runs indeed much faster.

I have just two more minor comments:
- Could you rename the con_vimmpi files to con_vimppi in the leaderboard folders?
- Could you in those two files change the short_description in the leaderboard_config dictionaries? Currently it is the copied description from the iLQR controller.

After you make these changes, I can merge the pull request to the main repo :)

domrachev03 commented:
Hello @fwiebe!

Thank you for the comments, we fixed those! We believe now it is ready to merge!

fwiebe commented:
Thanks, I will merge it now.
